### PR TITLE
fix: update document title for client route

### PR DESCRIPTION
**Pull request by tltv**
Updates document title for client view with flow layout based on `ViewConfig.title` and when not overridden by server's `HasDynamicTitle`.

RelatedTo: #20107

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java [hunk 1]
```diff
@@ -964,7 +965,11 @@ private static void updatePageTitle(NavigationEvent navigationEvent,
                 .filter(HasDynamicTitle.class::isInstance)
                 .map(tc -> ((HasDynamicTitle) tc).getPageTitle())
                 .filter(Objects::nonNull).findFirst()
-                .orElseGet(lookForTitleInTarget);
+                .orElseGet(() -> Optional
+                        .ofNullable(
+                                MenuRegistry.getClientRoutes(true).get(route))
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java:970]: Are you sure that the `route` has the same format as what `MenuRegistry` returns regarding slash in the beginning and end, e.g. `/foo` vs `foo` ? 
Maybe it's safer to trim slashes ?

Comment (tltv) [reply flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java:970]: I trust that it's safe here as it's coming from `handle` method where it's used in the same way.

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java:970]: Note that `handle` does 
```
        String route = event.getLocation().getPath().isEmpty()
                ? event.getLocation().getPath()
                : event.getLocation().getPath().startsWith("/")
                        ? event.getLocation().getPath()
                        : "/" + event.getLocation().getPath();
```
before checking the route.

**Review by mshabarov — APPROVED** [review]
Works as expected, LGTM

File: flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java (hunk 1 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java:970]: Also that is exactly the route string that is used.

**Pull request merged**